tools/netqtop_example.txt

```diff
@@ -0,0 +1,139 @@
+Demonstrations of netqtop.
+
+
+netqtop traces the kernel functions performing packet transmit (xmit_one) 
+and packet receive (__netif_receive_skb_core) on data link layer. The tool 
+not only traces every packet via a specified network interface, but also accounts 
+the PPS, BPS and average size of packets as well as packet amounts (categorized by 
+size range) on sending and receiving direction respectively. Results are printed 
+as tables, which can be used to understand traffic load allocation on each queue 
+of interested network interface to see if it is balanced. And the overall performance 
+is provided in the buttom.
+
+For example, suppose you want to know current traffic on lo, and print result 
+every second:
+# ./netqtop.py -n lo
+Fri Jul 31 16:00:01 2020
+TX
+ QueueID      BPS        PPS      avg_size   [0, 64)   [64, 512)  [512, 2K)  [2K, 16K)  [16K, 64K)
+    0       784.0        9.0         87          0          9          0          0          0
+ Total      784.0        9.0         87          0          9          0          0          0
+
+RX
+ QueueID      BPS        PPS      avg_size   [0, 64)   [64, 512)  [512, 2K)  [2K, 16K)  [16K, 64K)
+    0       658.0        9.0         73          4          5          0          0          0
+ Total      658.0        9.0         73          4          5          0          0          0
+----------------------------------------------------------------------------------------------------
+Fri Jul 31 16:00:02 2020
+TX
+ QueueID      BPS        PPS      avg_size   [0, 64)   [64, 512)  [512, 2K)  [2K, 16K)  [16K, 64K)
+    0        1.0K        3.0        341          0          2          1          0          0
+ Total       1.0K        3.0        341          0          2          1          0          0
+
+RX
+ QueueID      BPS        PPS      avg_size   [0, 64)   [64, 512)  [512, 2K)  [2K, 16K)  [16K, 64K)
+    0       982.0        3.0        327          1          1          1          0          0
+ Total      982.0        3.0        327          1          1          1          0          0
+----------------------------------------------------------------------------------------------------
```

Comment:
My screen looks like
```
    2       119.0        1.0        119          0          1          0          0          0
    3         0.0        0.0          0          0          0          0          0          0
    4         0.0        0.0          0          0          0          0          0          0
    5       239.0        2.0        119          0          2          0          0          0
    6       4.03K       11.0        375          0         10          1          0          0
    7         0.0        0.0          0          0          0          0          0          0
 Total      8.96K       20.0        458          0         13          7          0          0
----------------------------------------------------------------------------------------------------
Thu Aug 13 21:06:18 2020
TX
 QueueID      BPS        PPS      avg_size   [0, 64)   [64, 512)  [512, 2K)  [2K, 16K)  [16K, 64K)
    0       172.0        2.0         86          0          2          0          0          0
    1       1.92K        7.0        280          0          5          2          0          0
    2        86.0        1.0         86          0          1          0          0          0
    3       4.62K        3.0      1.54K          0          2          0          1          0
    4      24.55K       12.0      2.04K          0         10          0          1          1
    5       946.0       11.0         86          0         11          0          0          0
    6         0.0        0.0          0          0          0          0          0          0
    7       344.0        4.0         86          0          4          0          0          0
 Total     32.59K       40.0        834          0         35          2          2          1

RX
 QueueID      BPS        PPS      avg_size   [0, 64)   [64, 512)  [512, 2K)  [2K, 16K)  [16K, 64K)
    0         0.0        0.0          0          0          0          0          0          0
    1       628.0        2.0        314          0          2          0          0          0
    2       365.0        3.0        121          0          3          0          0          0
    3       224.0        2.0        112          0          2          0          0          0
    4       636.0        6.0        106          0          6          0          0          0
    5       374.0        3.0        124          0          3          0          0          0
    6       4.95K       13.0        390          0         11          2          0          0
    7         0.0        0.0          0          0          0          0          0          0
 Total     10.18K       33.0        315          0         27          6          0          0
----------------------------------------------------------------------------------------------------
^C-bash-4.4$
```
    2       119.0        1.0        119          0          1          0          0          0
    3         0.0        0.0          0          0          0          0          0          0
    4         0.0        0.0          0          0          0          0          0          0
    5       239.0        2.0        119          0          2          0          0          0
    6       4.03K       11.0        375          0         10          1          0          0
    7         0.0        0.0          0          0          0          0          0          0
 Total      8.96K       20.0        458          0         13          7          0          0
----------------------------------------------------------------------------------------------------
Thu Aug 13 21:06:18 2020
TX
 QueueID      BPS        PPS      avg_size   [0, 64)   [64, 512)  [512, 2K)  [2K, 16K)  [16K, 64K)
    0       172.0        2.0         86          0          2          0          0          0
    1       1.92K        7.0        280          0          5          2          0          0
    2        86.0        1.0         86          0          1          0          0          0
    3       4.62K        3.0      1.54K          0          2          0          1          0
    4      24.55K       12.0      2.04K          0         10          0          1          1
    5       946.0       11.0         86          0         11          0          0          0
    6         0.0        0.0          0          0          0          0          0          0
    7       344.0        4.0         86          0          4          0          0          0
 Total     32.59K       40.0        834          0         35          2          2          1

RX
 QueueID      BPS        PPS      avg_size   [0, 64)   [64, 512)  [512, 2K)  [2K, 16K)  [16K, 64K)
    0         0.0        0.0          0          0          0          0          0          0
    1       628.0        2.0        314          0          2          0          0          0
    2       365.0        3.0        121          0          3          0          0          0
    3       224.0        2.0        112          0          2          0          0          0
    4       636.0        6.0        106          0          6          0          0          0
    5       374.0        3.0        124          0          3          0          0          0
    6       4.95K       13.0        390          0         11          2          0          0
    7         0.0        0.0          0          0          0          0          0          0
 Total     10.18K       33.0        315          0         27          6          0          0
----------------------------------------------------------------------------------------------------
^C-bash-4.4$
```
Looks like the previous screen is not cleared completely and shown at the top. Did you see such an issue?

Reply:
yes the previous screen is not cleared. I did not implement that. It just print out results every second.

Reply:
I noticed that other tools like funclatency, the previous output is not cleared either. thought this would be acceptable?
e.g. 
```
# ./funclatency -i 2 -d 10 vfs_read
Tracing 1 functions for "vfs_read"... Hit Ctrl-C to end.

     nsecs               : count     distribution
         0 -> 1          : 0        |                                        |
         2 -> 3          : 0        |                                        |
         4 -> 7          : 0        |                                        |
         8 -> 15         : 0        |                                        |
        16 -> 31         : 0        |                                        |
        32 -> 63         : 0        |                                        |
        64 -> 127        : 0        |                                        |
       128 -> 255        : 0        |                                        |
       256 -> 511        : 29477    |********************                    |
       512 -> 1023       : 454      |                                        |
      1024 -> 2047       : 57783    |****************************************|
      2048 -> 4095       : 57       |                                        |
      4096 -> 8191       : 117      |                                        |
      8192 -> 16383      : 281      |                                        |
     16384 -> 32767      : 13       |                                        |
     32768 -> 65535      : 1        |                                        |

     nsecs               : count     distribution
         0 -> 1          : 0        |                                        |
         2 -> 3          : 0        |                                        |
         4 -> 7          : 0        |                                        |
         8 -> 15         : 0        |                                        |
        16 -> 31         : 0        |                                        |
        32 -> 63         : 0        |                                        |
        64 -> 127        : 0        |                                        |
       128 -> 255        : 0        |                                        |
       256 -> 511        : 25471    |****************                        |
       512 -> 1023       : 283      |                                        |
      1024 -> 2047       : 61617    |****************************************|
      2048 -> 4095       : 59       |                                        |
      4096 -> 8191       : 133      |                                        |
      8192 -> 16383      : 281      |                                        |
     16384 -> 32767      : 9        |                                        |

...